internal/api/v1/user/merkleroots/merkleroots_sync.go

```diff
@@ -0,0 +1,105 @@
+package merkleroots
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/url"
+
+	bip32 "github.com/bitcoin-sv/go-sdk/compat/bip32"
+	goclienterr "github.com/bitcoin-sv/spv-wallet-go-client/errors"
+	"github.com/bitcoin-sv/spv-wallet-go-client/queries"
+	"github.com/bitcoin-sv/spv-wallet/models"
+	"github.com/go-resty/resty/v2"
+)
+
+// MerkleRootsRepository is an interface responsible for storing synchronized MerkleRoots and retrieving the last evaluation key from the database.
+type MerkleRootsRepository interface {
+	// GetLastMerkleRoot should return the Merkle root with the highest height from your memory, or undefined if empty.
+	GetLastMerkleRoot() string
+	// SaveMerkleRoots should store newly synced merkle roots into your storage;
+	// NOTE: items are sorted in ascending order by block height.
+	SaveMerkleRoots(syncedMerkleRoots []models.MerkleRoot) error
+}
+
+type Client struct {
```

Comment:
Why do we make here a new client? Shouln't this function generally be used like:
this is a pseudo code so don't take it too literal.
```
client := spvwallet.NewClient()
err := client.SyncMerkleRoots
```

and expose this method in the `client.go`  as a part of it's api rather then making a new client? I don't see it made anywhere like this in this repo.

Reply:
whats wrong with making here client ?

Reply:
It's just weird to me that we have as you said a "general" client in package 'client' and here we also make another one. In tests you are not creating a general client but rather the one from this package 'merkleroots'. I think the merkleroots Sync method should be accessible from merkleRootsAPI which we already have rather than from separate client. 

But it's my just opinion maybe someone else could give their take on this.

Reply:
Allright it's just words :D 

Hopefully you know now what I meant. Please take a look at what @mgosek-4chain suggested [here](https://github.com/4chain-ag/spv-wallet-go-client-v2/pull/21#pullrequestreview-2456398165)

This is similar to what we had in the original version of go client and you could do something like

```
client, err := client.NewWithXPriv()
err = client.SetMerkleRootsDB(db)
client.SyncMerkleroots()
```

What do you think?